src/Cake.SqlServer/Backup/RestoreSqlBackupSettings.cs

```diff
@@ -42,12 +34,33 @@ public RestoreSqlBackupSettings()
         /// </summary>
         public bool WithReplace { get; set; }
 
+        /// <summary>
+        /// Before restoring backup, database will be switched to a single user mode. 
+        /// Default operation is to go into single user mode. However in some situation this might not work.
+        /// Use this switch to bypass single user mode and restore the DB as it is
+        /// 
+        /// This property has been obsoleted by SwitchToUserMode which allows for a third option
+        /// </summary>
+        [Obsolete("Use SwitchToUserMode instead")]
+        public bool SwitchToSingleUserMode
+        {
+            get => SwitchToUserMode == DbUserMode.SingleUser;
+            set => SwitchToUserMode = value ? DbUserMode.SingleUser : DbUserMode.MultiUser;
+        }
+
+        /// <summary>
+        /// Before restoring backup, database will be switched to the user mode selected
+        /// Default operation is to go into single user mode. However in some situation this might not work.
+        /// Use this switch to bypass single user mode and restore the DB as it is, or your can use restricted mode instead.
+        /// </summary>
+        public DbUserMode SwitchToUserMode { get; set; } = DbUserMode.SingleUser;
+
         /// <summary>
         /// Before restoring backup, database will be switched to a single user mode. 
         /// Default operation is to go into single user mode. However in some situation this might not work.
         /// Use this switch to bypass single user mode and restore the DB as it is
         /// </summary>
-        public bool SwitchToSingleUserMode { get; set; }
+        public bool SwitchToRestrictedUserMode { get; set; }
```

Comment:
As far as I can see this is not used anywhere and just been introduced. If we don't need it, can we remove it please?

Reply:
Fixed